docs/v3/core/index.rst

```diff
@@ -1598,46 +1466,232 @@ Storage transformers may be stacked to combine different functionalities:
+URLs during processing.
+
+- **Example:** ``https://example.com/zarr3/consolidated-metadata``
+- **Accepted regex:** ``^https?:\/\/[^/?#]+[^?#]*$``
+
+Extension versioning
+--------------------
+
+Extensions with **raw names** SHOULD follow the
+compatibility and versioning v3 `stability policy`_.
+
+For extensions with **URL-based names**, there are no guarantees in terms of
+versioning or compatibility. However, preserving backwards-compatibility is
+strongly encouraged.
+
+Extension example
+-----------------
+
+The following example of array metadata demonstrates these extension naming schemes::
+
+    {
+        "zarr_format": 3,
+        "data_type": "string", // raw name, short-hand name
+        "chunk_key_encoding": {
+            "name": "default", // core
+            "configuration": { "separator": "." }
+        },
+        "codecs": [
+            {
+                "name": "vlen-utf8" // raw name
+            },
+            {
+                "name": "zstd", // raw name
+                "configuration": { ... }
+            }
+        ],
+        "chunk_grid": {
+            "name": "regular", // core
+            "configuration": { "chunk_shape": [ 32 ] }
+        },
+        "shape": [ 128 ],
+        "dimension_names": [ "x" ],
+        "attributes": { ... }
+    }
+
+Extension specifications
+------------------------
+
+Extensions SHOULD have a published specification. A published specification
+facilitates multiple implementations of an extension.
+
+For extensions with raw names, the `zarr-extensions`_ repository
+SHOULD either contain the specification or link to it.
+
+For extensions with URL-based names, it is RECOMMENDED that the URL resolve to
+a specification of the extension. Additionally, URL-based extensions MAY also register
+themselves under the `zarr-extensions`_ repository for better discovery.
+
+Because the ``fill_value`` metadata key is dependent on the data type, 
+extension data types SHOULD specify permitted values for the ``fill_value`` in
+their specification.
+
+
+.. _extension-guidance:
+
+Guidance for extension authors
+------------------------------
+
+*This section is non-normative and provides assistance for the authors of
+extensions, especially those who are just getting started.*
+
+* If you are just getting started, use the URL of your work-in-progress as an
+  identifier for your extension. The GitHub link, including the branch if you
+  would like, makes a fine choice. This says to the community that this is a
+  draft, and if they are interested in the details, they can follow the URL to
+  find out more.
+
+* When developing an extension for which you intend to register a short name,
+  you may wish to test it using the short name even before you have registered
+  it. However, you MUST register the name before using the extension for
+  non-test purposes/for purposes where interoperability with other
+  implementations/users is a concern.
```

Comment:
```suggestion
* If you intend to distribute data widely using your extension, you SHOULD register your extension using Raw name, rather than a namespaced name.
```